src/Interceptor/Interceptor/EventInterceptor.php

```diff
@@ -15,6 +17,13 @@
 
 final class EventInterceptor extends AbstractInterceptor implements SuffixInterceptor, PrefixInterceptor
 {
+    public function __construct(
+        iterable $handlers = [],
+        private ?GenericEventHandler $genericEventHandler = null,
```

Comment:
I'm not sure if this needed, since you can use the same handler configure for the attribute, I don't see that case where you need to select two different handlers one for the use case event and one for the generic event.

In some extreme cases we can still add two times the annotation with different handlers

Reply:
the others handler are tight to an annotation or an attribute. the generic event will be sent even if there is no annotation or attribute defined on the use case.
and the event is different it need to contains an eventName `use_case.post.execute`the generic event name, and the name of the class that triggered the event. where the "standard" event will calculate the eventName regarding the class name that possess the annotation

Reply:
I see pretty well thought, we just need to be sure we don't duplicate event's as a lot of them use the old annotation